xbmc/pvr/PVRGUIInfo.cpp

```diff
@@ -747,7 +747,14 @@ void CPVRGUIInfo::UpdateBackendCache(void)
 
   // Update the backend information for all backends once per iteration
   if (m_iCurrentActiveClient == 0)
-    m_backendProperties = g_PVRClients->GetBackendProperties();
+  {
+    std::vector<SBackend> backendProperties;
+    {
+      CSingleExit exit(m_critSection);
```

Comment:
Just a quick question, is this the same as using `lock.Leave()` followed by `lock.Enter()`?